### PR TITLE
SEO description fallback to page description

### PR DESCRIPTION
**Pull request by steven1046**
Add a default description for meta tags when no seo-description is provided and the page has a description.

Title will be seo title, page title appended by site title, or fallback to the site title only in that order.
Description will be seo description, page description, or fallback to the site description only in that order.

#### Were the changes tested on
- [ ] Firefox (Desktop)
- [ ] Safari (Desktop)
- [x] Chrome (Desktop)
- [ ] Internet Explorer (Desktop)
- [ ] iOS (Mobile)
- [ ] Android (Mobile)
#### Running validation tools
- [ ] https://validator.w3.org/checklink
- [ ] https://validator.w3.org
- [ ] Dymanic Accessability Plugin (DAP)

### CODE REVIEW

**Comment by steven1046** [comment]
@kinueng the requested changes have been made.

We did not update the Twitter title to use the same title because we don't want to append the site title to the page title as it may be too long.

File: src/main/content/_includes/head.html [hunk 1]
```diff
@@ -1,11 +1,19 @@
-<head>
+{% assign description = '' %}
```
Comment (kinueng) [inline src/main/content/_includes/head.html]: Please format the tabs in this block of new code.  The alignment of the code is not lined up and can be confusing what line is nested in what code block.

File: src/main/content/_includes/head.html [hunk 2]
```diff
@@ -1,11 +1,19 @@
...
+  {% if page.seo-description %}    
+    {% assign description = page.seo-description | escape %}
```
Comment (kinueng) [inline src/main/content/_includes/head.html]: I'm going a bit back and forth with myself whether we need to `escape` our descriptions.  On one hand, `escape` will make our descriptions less human readable.  On the other hand, `escape` is good security practice for special characters.  @steven1046 your thoughts?

I also, oddly, could not actually get `escape` to work.  When I inspect the description, the special characters are not "encoded".  I used my local test environment and added characters `??` to `:page-seo-description:` to one of the guides.  I inspected the HTML and still see the non-escaped `??` in the `description` 

image

**Comment by steven1046** [comment]
@kinueng is there anything else to change for this pull request since we want to keep the liquid escape?

**Pull request merged**